nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,365 @@
+            date-field="dc:created"
+            grouped-by="ecm:primaryType"
+            where='{"ecm:hasLegalHold": "true"}'
+            group-limit="20"
+            data="{{legalHoldByDocType}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <nuxeo-card heading="[[i18n('retentionAnalytics.legalHoldByDocType')]]">
+            <chart-pie
+              values="[[_values(legalHoldByDocType)]]"
+              labels="[[_labels(legalHoldByDocType)]]"
+              options="{_piechartOptions}"
+            >
+            </chart-pie>
+          </nuxeo-card>
+        </div>
+
+        <!-- Expiration next 12 months -->
+        <nuxeo-repository-data
+          start-date="[[_formatDateNow()]]"
+          end-date="[[_formatDateFutureInMonths(12)]]"
+          with-date-intervals="month"
+          date-field="ecm:retainUntil"
+          data="{{expirationPerMonth}}"
+          index="[[index]]"
+        >
+        </nuxeo-repository-data>
+
+        <nuxeo-card class="chartBar" heading="[[i18n('retentionAnalytics.expirationNext12Months.heading')]]">
+          <chart-bar
+            labels="[[_futureExpirationLabels(expirationPerMonth)]]"
+            values="[[_values(expirationPerMonth)]]"
+            options='{ "legend": { "display": false }, "animation": false }'
+          >
+          </chart-bar>
+        </nuxeo-card>
+      </nuxeo-page>
+    `;
+  }
+
+  static get is() {
+    return 'nuxeo-retention-analytics';
+  }
+
+  static get properties() {
+    return {
+      index: {
+        type: String,
+        value: '_all',
+      },
+      visible: Boolean,
+      startDate: {
+        type: String,
+        value: moment().subtract(1, 'year').format('YYYY-MM-DD'),
+      },
+      endDate: {
+        type: String,
+        value: moment().format('YYYY-MM-DD'),
+      },
+      _filterActive: {
+        type: Array,
+        value: [{ range: { 'ecm:retainUntil': { gte: moment().format('YYYY-MM-DD') } } }],
+      },
+      _filterExpired: {
+        type: Array,
+        value: [{ range: { 'record:retainUntil': { lte: moment().format('YYYY-MM-DD') } } }],
+      },
+      _filterInterminate: {
+        type: Array,
+        value: [{ range: { 'ecm:retainUntil': { gte: '9999-01-01T00:00:00.000Z' } } }],
+      },
+      _labelsRetentionStatus: {
+        type: Array,
+        value: ['No retention', 'Active retention', 'Expired retention', 'Interminate retention'],
+      },
+      _piechartOptions: {
+        type: Object,
+        value: { legend: { display: true, position: 'bottom', labels: { boxWidth: 12 } }, animation: false },
+      },
+    };
+  }
+
+  static get observers() {
+    return ['_resetStartDate(visible)'];
+  }
+
+  _resetStartDate(visible) {
+    // we trigger the elasticsearch by modifying the startDate each time the page is visible
+    if (visible) {
+      this.startDate = moment(this.startDate).subtract(1, 'hour').format('YYYY-MM-DD');
```

Comment:
This way you will always decrease the starting date. see above 

https://user-images.githubusercontent.com/46494950/104934168-1a91b080-59aa-11eb-9c4a-900e0619ee22.mov


I have tried another approche, where you query all repositories data of the given elements (analytics one)

```
 const repositories = this.shadowRoot.querySelectorAll('nuxeo-repository-data');
 repositories.forEach((repository) => repository.ready());

```

I have looked to the `nuxeo-repository-data` element and I saw that it inherent from `nuxeo-aggregate-data-behavior` and that we have the read method [Here](https://github.com/nuxeo/nuxeo-elements/blob/master/dataviz/nuxeo-aggregate-data-behavior.js#L189)


can you try to add ⬆️ and we will also wait for more feedbacks from UI team on this approche.

Reply:
@RSalem07 thanks for the suggestion! indeed, that works and it looks like a better practice - I will adopt it.